src/crc32c/.ycm_extra_conf.py

```
@@ -0,0 +1,142 @@
+    file_path: The path to the source file. Should be inside the project. Used
+      to locate the relevant .clang_complete file.
+    build_root: The current directory when running the Clang compiler for this
+        file. Should be an absolute path.
+
+  Returns:
+    A list of strings, where each element is a Clang command-line flag.
+  """
+  clang_complete_path = FindNearest('.clang_complete', file_path, build_root)
+  if clang_complete_path is None:
+    return None
+  clang_complete_flags = open(clang_complete_path, 'r').read().splitlines()
```

Ruff (0.8.0)

Use a context manager for opening files

(SIM115)